php-checks/src/main/java/org/sonar/php/checks/UndocumentedApiCheck.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import com.google.common.io.Files;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.php.api.CharsetAwareVisitor;
+import org.sonar.plugins.php.api.tree.declaration.ClassDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.FunctionDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.MethodDeclarationTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+
+/**
+ * Checks every method and class declaration for presence of a comment in the line before
+ * <p>
+ * Every declaration should have a short comment explaining what the {class|method|function} is doing
+ *
+ * @author Nils-Janis Mahlstädt <nils-janis.mahlstaedt@hmmh.de>
+ */
+@Rule(
+        key = UndocumentedApiCheck.KEY,
+        name = UndocumentedApiCheck.MESSAGE,
+        priority = Priority.MINOR,
+        tags = {Tags.DESIGN, Tags.CONVENTION})
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.UNDERSTANDABILITY)
+@SqaleConstantRemediation("10min")
+public class UndocumentedApiCheck extends PHPVisitorCheck implements CharsetAwareVisitor {
+
+    public static final String KEY = "S1176";
+    public static final String MESSAGE = "method, function and class declarations should always be documented";
+
+    private static final String MESSAGE_PARTIAL = " declaration should always be documented";
+    private static final boolean defaultTrue = true;
+    private static final boolean defaultFalse = false;
+
+
+    private Charset charset;
+    private String currentFilename = "";
+    //lines in currentFile
+    private List<String> internal_lines;
+
+    // check if class has comment
+    @RuleProperty(
```

Comment:
Hello Nils,

I don't get your question. All parameters for this rule (there are 8 of them) are boolean, either the rule should force some doc to appear, either not.